tests/test_datediff_level.py

```diff
@@ -1,384 +1,188 @@
-import logging
+from datetime import datetime
 
-import pandas as pd
 import pytest
 
 import splink.comparison_level_library as cll
 import splink.comparison_library as cl
-from splink import exceptions
 from splink.column_expression import ColumnExpression
+from tests.decorator import mark_with_dialects_excluding, mark_with_dialects_including
+from tests.literal_utils import (
+    ComparisonLevelTestSpec,
+    ComparisonTestSpec,
+    LiteralTestValues,
+    run_tests_with_args,
+)
 
-from .decorator import mark_with_dialects_excluding
 
-
-# No SQLite - no array comparisons in library
 @mark_with_dialects_excluding("sqlite")
-def test_datediff_levels(test_helpers, dialect):
+def test_absolute_time_difference_levels_date(test_helpers, dialect):
+    # Check it's fine to pass a date into this function
     helper = test_helpers[dialect]
-
-    # Capture differing comparison levels to allow unique settings generation
-    df = pd.DataFrame(
-        [
-            {
-                "unique_id": 1,
-                "first_name": "Tom",
-                "dob": "2000-01-01",
-            },
-            {
-                "unique_id": 2,
-                "first_name": "Robin",
-                "dob": "2000-01-30",
-            },
-            {
-                "unique_id": 3,
-                "first_name": "Zoe",
-                "dob": "1995-09-30",
-            },
-            {
-                "unique_id": 4,
-                "first_name": "Sam",
-                "dob": "1966-07-30",
-            },
-            {
-                "unique_id": 5,
-                "first_name": "Andy",
-                "dob": "1996-03-25",
-            },
-            {
-                "unique_id": 6,
-                "first_name": "Alice",
-                "dob": "2000-03-25",
-            },
-            {
-                "unique_id": 7,
-                "first_name": "Afua",
-                "dob": "1960-01-01",
-            },
-        ]
+    db_api = helper.extra_linker_args()["database_api"]
+
+    col_exp = ColumnExpression("dob").try_parse_date()
```

Comment:
This is how I ensure the value going into the test is a true date type (the cast from string to date is in the innermost brackets)